src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -984,7 +984,7 @@ private long GetAssemblyId(WriteObjectInfo objectInfo)
             {
                 assemId = 0;
             }
-            else if (assemblyString.Equals(Converter.s_urtAssemblyString))
+            else if (assemblyString.Equals(Converter.s_urtAssemblyString) || assemblyString.Equals(Converter.s_urtAlternativeAssemblyString))
```

Comment:
I was thinking about that for a while now and concluded that using the TypeForwardedFrom attributes directly which means teaching the stripping optimization logic to use the "correct" FullQualifiedName makes sense and we can skip resolving via the mscorlib shim which would also improve performance. The TypeForwardedFrom attributes are the "source" for the generated mscorlib shim so it just make sense to directly look at them...